docs/pages/userdocs/install.dox

```diff
@@ -0,0 +1,130 @@
+ * multi-configuration generator:
+ *
+ * \code{.sh}
+ * cmake --install build --config Release
+ * \endcode
+ *
+ *
+ *
+ * \section devbuild_sec Developer Build
+ *
+ * Build system targets that are only useful for developers of AqNWB are
+ * hidden if the `aq-nwb_DEVELOPER_MODE` option is disabled. Enabling this
+ * option makes tests and other developer targets and options available. You can enable
+ * the option when configuring the build by adding `-Daq-nwb_DEVELOPER_MODE=ON`, e.g.,
+ *
+ * \code{.sh}
+ * cmake -S . -B build -Daq-nwb_DEVELOPER_MODE=ON
+ * \endcode
+ *
+ * \subsection  devbuild_presets_subsec Presets
+ *
+ * As a developer, you can create your own dev preset by making a `CMakeUserPresets.json` file at the root of
+ * the project:
+ *
+ * \code{.json}
+ * {
+ *   "version": 2,
+ *   "cmakeMinimumRequired": {
+ *     "major": 3,
+ *     "minor": 15,
+ *     "patch": 0
+ *   },
+ *   "configurePresets": [
+ *     {
+ *       "name": "dev",
+ *       "binaryDir": "${sourceDir}/build/dev",
+ *       "inherits": ["dev-mode", "ci-<os>"],
+ *       "cacheVariables": {
+ *         "CMAKE_BUILD_TYPE": "Debug"
+ *       }
+ *     }
+ *   ],
+ *   "buildPresets": [
+ *     {
+ *       "name": "dev",
+ *       "configurePreset": "dev",
+ *       "configuration": "Debug"
+ *     }
+ *   ],
+ *   "testPresets": [
+ *     {
+ *       "name": "dev",
+ *       "configurePreset": "dev",
+ *       "configuration": "Debug",
+ *       "output": {
+ *         "outputOnFailure": true
+ *       }
+ *     }
+ *   ]
+ * }
+ * \endcode
+ * Replace `<os>` in the `CMakeUserPresets.json` file with the name of
+ * the operating system you have (`win64`, `linux` or `darwin`).
+ *
+ * \subsection configure_build_test Configure, build and test
+ *
+ * You can configure, build and test the project respectively with the following commands from the project root on
+ * any operating system with any build system:
+ *
+ * \code{.sh}
+ * cmake --preset=dev
+ * cmake --build --preset=dev
+ * ctest --preset=dev
+ * \endcode
+ *
+ * \subsection devbuild_dev_mode_targets_subsec Developer mode targets
+ *
+ * Additional targets can be invoked when in development mode using the commands below
+ *
+ * \code{.sh}
+ * cmake --build --preset=dev --target=<name of the target>
+ * \endcode
+ *
+ * \subsubsection  devbuild_target_options_subsubsec Target options
+ * - `format-check`: run the `clang-format` tool on the codebase to check for formatting errors
+ * - `format-fix` :  run the `clang-format` tool on the codebase with `FIX=YES` to both check and automatically fix for formatting errors
+ * - `spell-check`: run the `codespell` tool on the codebase to check for common spelling errors
+ * - `spell-fix` :  run the `codespell` tool on the codebase with `FIX=YES` to both check and automatically fix  common spelling errors
+ * - `docs` : builds the documentation using Doxygen. (Note: run `cmake --preset=dev -DBUILD_DOCS=ON` before building to add docs target)
+ */
+
+
+
+
+ */
```

Comment:
The way I considered it, a User is someone using AqNWB in their code (e.g.,to integrate NWB with their system) and a developer is someone contributing to AqNWB. Since we are in C++, both users and developers need to know how to build the library (or user at least need to be able to link to it if we do binary releases). We could split the page into two parts to have separate install pages in the user and developer sections.

Reply:
I have separated the installation instructions for the users and developers on respective pages in the userdocs/devdocs as you suggested. I've moved the requirements back to the main page so that we can reference them from both installation docs and avoid duplication.